src/main/java/net/ripe/rpki/rsyncit/rrdp/RrdpFetcher.java

```diff
@@ -214,6 +221,19 @@ private ProcessPublishElementResult processPublishElements(Element doc) throws X
         return new ProcessPublishElementResult(objects, collisionCount.get());
     }
 
+    /**
+     * Generate timestamp that will be tracked per object and used as FS modification timestamp.
+     * Use last-modified header from the snapshot if accessible, otherwise truncate current time
+     * to the closest hour -- it is unlikely that different instances will have clocks off by a lot,
+     * so rounding down to an hour should generate the same timestamps _most of the time_.
+     */
+    private Instant makeTimestampForObjects(Instant lastModified) {
+        if (lastModified != null) {
```

Comment:
The instance will be initialised to `0`, not null in the code that takes the header. The fallback path will never be hit (instead we fill set files to 1970)?